src/python/graplctl/graplctl/aws_cdk_ops.py

```diff
@@ -0,0 +1,120 @@
+import base64
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+from typing import IO, AnyStr
+
+from mypy_boto3_lambda import LambdaClient
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(os.getenv("GRAPL_LOG_LEVEL", "INFO"))
+LOGGER.addHandler(logging.StreamHandler(stream=sys.stdout))
+
+
+def deploy_grapl(
+    grapl_root: pathlib.Path, aws_profile: str, stdout: IO[AnyStr], stderr: IO[AnyStr]
+) -> None:
+    grapl_cdk_dir = pathlib.Path(grapl_root.absolute(), "src/js/grapl-cdk")
+    edge_ux_artifact_dir = pathlib.Path(grapl_cdk_dir, "/edge_ux_post_replace")
+    outputs_file = pathlib.Path(grapl_cdk_dir, "cdk-output.json")
+
+    if edge_ux_artifact_dir.exists():
+        shutil.rmtree(edge_ux_artifact_dir)
+
+    os.mkdir(edge_ux_artifact_dir)
+
+    LOGGER.info("building cdk")
+    subprocess.run(
+        f"cd {grapl_cdk_dir.as_posix()} && npm run build",
+        stdout=stdout,
+        stderr=stderr,
+        check=True,
+        shell=True,
+    )
+    LOGGER.info("built cdk")
+
+    LOGGER.info("deploying Grapl stack")
+    subprocess.run(
+        f"cd {grapl_cdk_dir.as_posix()} && AWS_PROFILE={aws_profile} cdk deploy --require-approval=never --profile={aws_profile} --outputs-file={outputs_file.as_posix()} Grapl",
```

Comment:
All these `subprocess.run` calls could be simplified a bit by using the `cwd` keyword argument to specify the working directory, rather than using the `cd <directory> && <command>` pattern.

Similarly, rather than embedding environment setup in the command string, we could use the `env` keyword argument.

That, in turn, could pave the way for a bit of abstraction on these command invocations by allowing you to more easily create a helper function to run things, since each invocation is largely structurally the same (they all take place in the same directory, with the same environment, with the same keyword argument values)

Reply:
👍 will fix

Reply:
Done. This refactor appears to have broken colorized terminal output, I'm not sure why.